### PR TITLE
Add TypeScript support, bump to babel v7

### PR DESCRIPTION
**Pull request by williamboman**
I've included stuff from #505.

### CODE REVIEW

**Comment by williamboman** [comment]
@lencioni This should be ready for 👀 now

File: lib/parse.js [hunk 1]
```diff
@@ -1,18 +1,42 @@
-export default function parse(fileContent: String): Object {
-  // For some reason, we can't `import` babylon using an es6 import.
+const TYPESCRIPT_FILE_PATH_REGEX = /\.tsx?$/
+
+export default function parse(fileContent: String, absolutePathToFile: String): Object {
```
Comment (williamboman) [inline lib/parse.js]: Not the cleanest function signature (adding `absolutePathToFile: String`). I'm too lazy to come up with a better solution. Ideally I feel like this module should not incorporate any concept of files at all.

File: package.json [hunk 2]
```diff
@@ -43,33 +45,36 @@
   },
   "homepage": "https://github.com/galooshi/import-js#readme",
   "devDependencies": {
-    "babel-cli": "^6.26.0",
-    "babel-core": "^6.26.0",
-    "babel-eslint": "^8.0.0",
-    "babel-jest": "^21.0.2",
+    "@babel/cli": "^7.0.0-beta.51",
```
Comment (lencioni) [inline package.json]: Do you mind bumping this to the latest beta? (currently beta 55 it seems)

File: lib/parse.js (hunk 1 above)
Comment (lencioni) [reply lib/parse.js]: Agreed. I wonder if this would be better as an options object with a key of `typePlugins`?

File: lib/parse.js [hunk 3]
```diff
@@ -1,18 +1,42 @@
...
+  // For some reason, we can't `import` @babel/parser using an es6 import.
   // eslint-disable-next-line global-require
-  const babylon = require('babylon');
+  const babelParser = require('@babel/parser');
+
+  const typePlugins = TYPESCRIPT_FILE_PATH_REGEX.test(absolutePathToFile)
+    ? ['typescript']
+    : ['flow']
```
Comment (lencioni) [inline lib/parse.js]: Not a big deal either way, but why use an array here instead of a string?

File: lib/parse.js (hunk 1 above)
Comment (williamboman) [reply lib/parse.js]: That's exactly what I thought as well, and is what I did initially. However, I opted for the more rudimentary solution as I feel an options object parameter just introduces bloat and will make things harder to refactor. I'm up for either though 🤷‍♂️

File: lib/parse.js (hunk 3 above)
Comment (williamboman) [reply lib/parse.js]: Just lazy coding 🙈, was unsure how many plugins was required for TS so I coded it as an array before figuring it out.

File: yarn-error.log [hunk 4]
```diff
@@ -0,0 +1,131 @@
+Arguments:
```
Comment (lencioni) [inline yarn-error.log]: It looks like an error log snuck in. Mind removing this from your PR?

Comment (williamboman) [reply yarn-error.log]: Woops, done. Added some stuff to `.gitignore` while I was at it

File: package.json [hunk 5]
```diff
@@ -43,33 +45,36 @@
   },
   "homepage": "https://github.com/galooshi/import-js#readme",
   "devDependencies": {
-    "babel-cli": "^6.26.0",
-    "babel-core": "^6.26.0",
-    "babel-eslint": "^8.0.0",
-    "babel-jest": "^21.0.2",
+    "@babel/cli": "^7.0.0-beta.55",
```
Comment (lencioni) [inline package.json]: It looks like beta 56 was published last week. Mind updating again?

Comment (williamboman) [reply package.json]: The semver range will include it and any later beta releases

Comment (trotzig) [reply package.json]: I still got beta.55 when I installed. Are you sure semver ranges work with pre-releases like these? Nothing to worry about though, I'm going to bump us to rc.1 anyway. :)

Comment (williamboman) [reply package.json]: It should, it should actually also include release candidates.

**Pull request merged**